src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -154,72 +181,162 @@ export default class SimidMapCreative extends BaseSimidCreative {
+      center: coordinates
+    });
+    new google.maps.Marker({
+      position: coordinates,
+      map: this.map_,
+      title: 'Current Position'
+    });
+    this.findNearby_(this.searchQuery_, coordinates);
+  }
 
-/**
- * Searches for the closest corresponding businesses based off of the given search parameter,
- * and places pins on the map that represent the 4 closest locations.
- * @param {String} searchParameter A string with the business's name to use in the query.
- * @param {!google.maps.LatLng} coordinates The LatLng object of user's current location.
- * @private 
- */
-findNearby_(searchParameter, coordinates) {
-  const request = {
-    location: coordinates,
-    name: searchParameter,
-    openNow: true,
-    rankBy: google.maps.places.RankBy.DISTANCE
-  };
-  const service = new google.maps.places.PlacesService(this.map_);
-  service.nearbySearch(request, this.displayResults_.bind(this));
-}
+  /**
+   * Searches for the closest corresponding locations based off of the given search parameter,
+   * and places pins on the map that represent the 4 closest locations.
+   * @param {String} searchParameter A string with the advertiser's name to use in the query.
+   * @param {!google.maps.LatLng} coordinates The LatLng object of user's current location.
+   * @private 
+   */
+  findNearby_(searchParameter, coordinates) {
+    const request = {
+      location: coordinates,
+      name: searchParameter,
+      openNow: true,
+      rankBy: google.maps.places.RankBy.DISTANCE
+    };
+    const service = new google.maps.places.PlacesService(this.map_);
+    service.nearbySearch(request, this.displayResults_.bind(this));
+    this.createTravelChoices_();
+  }
 
-/**
- * Displays the closest business locations to a user's current location.
- * @param {!Object} results An array of Place Results from the search query.
- * @param {!google.maps.places.PlacesServiceStatus} status The status returned 
- *  by the PlacesService on the completion of its searches.
- * @private 
- */
-displayResults_(results, status) {
+  /**
+   * Displays the closest advertisement's locations to a user's current location.
+   * @param {!Object} results An array of Place Results from the search query.
+   * @param {!google.maps.places.PlacesServiceStatus} status The status returned 
+   *  by the PlacesService on the completion of its searches.
+   * @private 
+   */
+  displayResults_(results, status) {
     if (status == google.maps.places.PlacesServiceStatus.OK) {
       for (let i = 0; i < DEFAULT_LOCATION_NUM_DISPLAYED; i++) {
         this.placeMapMarker_(results[i]);
       }
+      this.activeLocation_ = results[0].geometry.location;
+      this.displayDirections_(this.activeLocation_, this.currentLocation_);
+    } else {
+      const statusErrorMessage = {
+        message: "ERROR: Places Service Status was: "+status,
+      };
+      this.simidProtocol.sendMessage(CreativeMessage.LOG, statusErrorMessage);
+      this.playContent_();
```

Comment:
if the map errors out, wouldn't it make sense to go back to the ad - since they were interested enough to check the map? this is okay as well though

Reply:
I agree with this so return to the ad and then continue to display the skip ad button

Reply:
Perhaps change message to 'ERROR: Failed to load map: + status "

Reply:
Changed to "ERROR: Failed to complete search: + status"